src/ksym.rs

```diff
@@ -91,29 +95,25 @@ impl KSymResolver {
         }
     }
 
-    pub fn find_address_ksym(&self, addr: u64) -> Option<&Ksym> {
+    pub fn find_addresses_ksym(&self, addr: u64) -> impl Iterator<Item = &Ksym> {
         let mut l = 0;
         let mut r = self.syms.len();
 
-        if !self.syms.is_empty() && self.syms[0].addr > addr {
-            return None;
-        }
-
-        while l < (r - 1) {
-            let v = (l + r) / 2;
-            let sym = &self.syms[v];
+        while l < r {
+            let m = (l + r) / 2;
+            let sym = &self.syms[m];
 
-            if sym.addr == addr {
-                return Some(sym);
-            }
             if addr < sym.addr {
-                r = v;
+                r = m;
             } else {
-                l = v;
+                l = m + 1;
             }
         }
-
-        Some(&self.syms[l])
+        // `self.syms[l - 1].addr <= addr < self.syms[l].addr`.
+        self.syms[0..l]
+            .iter()
+            .rev()
+            .take_while(move |sym| sym.addr == self.syms[l - 1].addr)
```

Comment:
The test case is already added.
if `l` is 0, the closure will never be executed since self.syms[0..0] is empty.  If `l` is `1`, it will compare to itself. I means `sym` is identical to `self.syms[l - 1]`.